server/routes/soonBusStop.js

```diff
@@ -0,0 +1,126 @@
+var calcCWorCCW = require('./direction.js')
+const busStops = require('./bus-stops.json')
+const defaultDatabase = require('./firebase.js')
+
+// updates next 3 bus stops for every bus
+module.exports = function nextBusStops() {
+  const busCollection = [];
+  let stops_arr_CW = [];
+  let stops_arr_CCW = [];
+
+  // Push all bus ID's into array
+  defaultDatabase.collection('busses').get().then(function(querySnapshot) {
```

Comment:
There will be a problem here. Since the .then statement is asynx, you are not waiting for this operation to complete before moving on to the next parts of your code. It means that your busCollection might still be empty when you start using it later on. You might want to consider using async/await to handle this: